src/main/java/core/basesyntax/service/reader/ReadFileFruitImpl.java

```
@@ -0,0 +1,23 @@
+package core.basesyntax.service.reader;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ReadFileFruitImpl implements ReadFileFruit {
+    @Override
+    public List<String> readAll(String filePath) {
+        List<String> list = new LinkedList<>();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                list.add(line);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

The catch block currently rethrows the exception as `new RuntimeException(e)` which lacks an informative message. The checklist states: "Don't ignore exceptions in try-catch blocks" and recommends rethrowing a RuntimeException with an informative message and the original exception (for example: `throw new RuntimeException("Can't find file by path: " + filePath, e);`). Update the catch to include `filePath` in the message and pass `e` as the cause so callers/tests can see what failed and where.